flow/Trace.cpp

```diff
@@ -585,6 +585,27 @@ bool traceFormatImpl(std::string& format) {
 		return false;
 	}
 }
+
+template <bool validate>
+bool traceClockSource(std::string& source) {
+	std::transform(source.begin(), source.end(), source.begin(), ::tolower);
+	if (source == "now") {
+		if (!validate) {
+			g_trace_clock = TRACE_CLOCK_NOW;
+		}
+		return true;
+	} else if (source == "realtime") {
+		if (!validate) {
+			g_trace_clock = TRACE_CLOCK_REALTIME;
+		}
+		return true;
+	} else {
+		if (!validate) {
+			g_trace_clock = TRACE_CLOCK_NOW;
```

Comment:
It may be better to leave this unchanged in this condition, just in case somebody did something like set the clock to `realtime` and then to `foo`.

Reply:
Seems reasonable to me. I was following the trace_format PR quite closely here without a more general understanding of the code base. In my head I was generally only thinking that this would run once before the network started but that check is further up which means we could eventually get to squirrelly issues if this got repurposed in the future.